aiida_hyperqueue/scheduler.py

```
@@ -57,7 +62,26 @@ def validate_resources(cls, **kwargs):
         try:
             resources.num_cpus = kwargs.pop("num_cpus")
         except KeyError:
-            raise KeyError("Must specify `num_cpus`")
+            try:
+                # For backward compatibility where `num_machines` and `num_mpiprocs_per_machine` are setting
+                # TODO: I only setting the default value as 1 for `num_mpiprocs_per_machine` because aiida-quantumespresso override
+                # resources default with `num_machines` set to 1 and then get builder with such setting.
+                # The `num_mpiprocs_per_machine` sometime can be read from "Default #procs/machine" of computer setup but if it is not exist
+                # the builder can not be properly get without passing `option` to builder generator.
+                # It is anyway a workaround for backward compatibility so this default is implemented despite it is quite specific for the qe plugin.
+                resources.num_cpus = kwargs.pop("num_machines") * kwargs.pop(
+                    "num_mpiprocs_per_machine", 1
+                )
+            except KeyError:
+                raise KeyError(
+                    "Must specify `num_cpus`, or (`num_machines` and `num_mpiprocs_per_machine`)"
+                )
+            else:
+                message = "The `num_machines` and `num_mpiprocs_per_machine` for setting hyperqueue resources are deprecated. "
+                "Please set `num_cpus` and `memory_mb`."
+
+                message = f"{message} (this will be removed in aiida-hyperqueue v1.0)"
+                warnings.warn(message, AiiDAHypereQueueDeprecationWarning, stacklevel=3)
```

If my previous comment is ture, here we need to raise the error message instead of warnning.

---

No the raise is in the `KeyError` conditional branch. Here the warning is from else branch which warns when the `num_cpus` is get from `num_mpiprocs_per_machine` and `num_machines` instead of directly set.